QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift

```diff
@@ -18,21 +18,31 @@ class ProgressView: UIView {
     }
 
     func set(colors: [UIColor]) {
-        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+        set(total: colors.count)
 
-        for index in 0..<colors.count {
-            if colors[index] == UIColor.white {
-                label.text = "\(index + 1)/\(colors.count)"
-            }
+        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
 
+        for color in colors {
             let cell = UIView()
-            cell.backgroundColor = colors[index]
+            cell.backgroundColor = color
             cell.layer.cornerRadius = 2
 
             stackView.addArrangedSubview(cell)
         }
     }
 
+    private func set(total: Int) {
+        guard let current = label.text?.components(separatedBy: "/")[0] else { return }
```

Comment:
Do not rely on values from UI components. Instead, pass this value in this func.